Core/GUI/GameTooltipHooks.lua

```diff
@@ -515,14 +515,20 @@ local function onTooltipSetItem(tooltip, tooltipData)
 		return
 	end
 
-	local _, itemLink = tooltip:GetItem()
-	if type(itemLink) ~= "string" then
+	local name, link = tooltip:GetItem()
+	if type(link) ~= "string" or not name then
 		return
 	end
 
-	local id = itemLink:match("item:(%d+):")
+	local id = link:match("item:(%d*)")
```

Comment:
I'm afraid I don't understand the reason behind this. Could you please clarify why the pattern should change?

AFAIK an item ID  must always have at least one digit. The problem would be the `keystone` prefix, no?